eng/Workarounds.targets

```diff
@@ -4,7 +4,7 @@
     <!-- Reference base shared framework at incoming dependency flow version, not bundled sdk version. -->
     <FrameworkReference
       Update="Microsoft.NETCore.App"
-      Condition=" '$(TargetFramework)' == '$(DefaultNetCoreTargetFramework)' AND '$(TargetLatestDotNetRuntime)' != 'false' "
+      Condition=" (('$(TargetFramework)' == '$(DefaultNetCoreTargetFramework)') OR ('$(TargetFramework)' == '$(DefaultNetCoreTargetFramework)-windows')) AND '$(TargetLatestDotNetRuntime)' != 'false' "
```

Comment:
Is this change related to https://github.com/dotnet/aspnetcore/pull/31817#issuecomment-819881555? Maybe add a comment to clarify it here...